beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestCatalogValidation.java

```diff
@@ -67,6 +65,26 @@ public void testUploadCatalog() throws Exception {
     	}
     }
     
+    @Test(groups = "slow", description = "https://github.com/killbill/killbill/issues/1465")
+    public void testUploadCatalogPlanValidation() throws Exception {
+	/*
+	* This is the test case for #1465. CatalogValidation-v3.xml is added to the DB even if the validation fails for 1465. So to test this #1465 validation, 2nd catalog CatalogValidation-v4.xml needs to be updated and it validates those validations.
```

Comment:
I don't understand the comment, and how it relates to the test, but the fact that the comment is misleading is not even the issue here. The issue is that your test does not do what it should be doing, and it does not check that the validation code is correct (and it's not correct). so, let's unpack everything:


In the first validation for `CatalogValidation-v3.xml`, you have a `standard-monthly` with an `EVERGREEN` final phase whose duration is *not* unlimited, so this is indeed a good test case for #1465. However, you don't have any assertion to verify that the upload failed, and as a result you are not hitting the assertion in the `catch` statement line 77. As a result, your catalog version gets uploaded.

The reason why the validation does not work is because of at least these 2 reasons:
* [this](https://github.com/killbill/killbill/blob/master/catalog/src/main/java/org/killbill/billing/catalog/api/user/DefaultCatalogUserApi.java#L109): Since this is the first version, `versionedCatalog` ends up being null, and you never enter the validation path.
* But there is more, even if you were to fix this, there is another issue [here](https://github.com/tejinderghatore/killbill/blob/master/catalog/src/main/java/org/killbill/billing/catalog/DefaultVersionedCatalog.java#L167): Your code only calls the `validatePlanShape` if there are multiple versions, which is true if you compare things across versions (#1481) but not true if you check things within one version (#1465)

Then, in the second validation for `CatalogValidation-v4.xml`, because of the previous issue**s**, there is now a version of the catalog uploaded with a plan `standard-monthly`, and so although  there is now a version of that same `Plan` with an `EVERGREEN` final phase whose duration is unlimited, it now fails - and it should not.